source/joker_effects.c

```diff
@@ -212,69 +238,75 @@ static JokerEffect joker_stencil_effect(Joker *joker, Card *scored_card) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
 
     effect.mult = random() % (MISPRINT_MAX_MULT + 1);
 
     return effect;
 }
 
-static JokerEffect walkie_talkie_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect walkie_talkie_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card == NULL)
-        return effect;
 
-    if (scored_card->rank == TEN || scored_card->rank == FOUR) {
-            effect.chips = 10;
-            effect.mult = 4;
-    }
+    SCORE_ON_CARD_SCORED_ONLY(scored_card, scored_when, effect)
 
+    if (scored_card && (scored_card->rank == TEN || scored_card->rank == FOUR)) {
+        effect.chips = 10;
+        effect.mult = 4;
+    }
+    
     return effect;
 }
 
-static JokerEffect fibonnaci_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect fibonnaci_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card == NULL)
-        return effect;
+
+    SCORE_ON_CARD_SCORED_ONLY(scored_card, scored_when, effect)
 
     switch (scored_card->rank) {
-        case ACE: case TWO: case THREE: case FIVE: case EIGHT:
+        case ACE:
+        case TWO:
+        case THREE:
+        case FIVE:
+        case EIGHT:
             effect.mult = 8;
+            break;
         default:
             break;
     }
 
     return effect;
 }
 
-static JokerEffect banner_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect banner_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
 
     effect.chips = 30 * get_num_discards_remaining();
 
     return effect;
 }
 
-static JokerEffect mystic_summit_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect mystic_summit_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
 
-    if (get_num_discards_remaining() == 0)
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
+
+    if (get_num_discards_remaining() == 0) {
         effect.mult = 15;
+    }
 
     return effect;
 }
 
-static JokerEffect blackboard_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect blackboard_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
```

Comment:
```suggestion
    SCORE_ON_EVENT_ONLY(JOKER_CALLBACK_INDEPENDANT, scored_when, effect)
```

Batch - Consolidate joker event restriction macros